src/2024h2/yank-crates-with-a-reason.md

```diff
@@ -0,0 +1,72 @@
+# Yank crates with a reason
+
+| Metadata |                      |
+| -------- | -------------------- |
+| Owner(s) | [hi-rustin]          |
+| Teams    | [crates.io], [Cargo] |
+| Status   | WIP                  |
+
+[hi-rustin]: https://github.com/hi-rustin
+[Cargo]: https://www.rust-lang.org/governance/teams/dev-tools#team-cargo
+[crates.io]: https://www.rust-lang.org/governance/teams/dev-tools#team-crates-io
+
+## Summary
+
+Over the next 6 months, we will first implement a feature in crates.io that asks for a reason when a administrator yanks a crate. After this feature has been up and running for a while, we'll open it up to Cargo to support filling in the reason for yanking. Make it an optional parameter of the registry yank API.
+
+## Motivation
+
+When a crate is updated to address a critical issue—such as a fix for a soundness bug or a security vulnerability—it is beneficial to yank previous versions and prompt users to upgrade with a yank reason. Additionally, if a crate is renamed or deprecated, the yank message can provide guidance on the new recommended crate or version. This ensures that users are aware of necessary updates and can maintain the security and stability of their projects.
+
+### The status quo
+
+We came up with [this need](https://github.com/rust-lang/cargo/issues/2608) eight years ago, but it was never implemented.
+
+This feature has the following potential use cases:
+
+1. When a crate is fixed because it will be broken in the next version of the compiler (e.g. a soundness fix or bug fix) then the previous versions can be yanked and nudge users forward.
+2. If a crate is fixed for a security reason, the old versions can be yanked and the new version can be suggested.
```

Comment:
I notice this differs from [the current recommendation of "when to yank"](https://doc.rust-lang.org/nightly/cargo/commands/cargo-yank.html#when-to-yank).
Regardless, this is why we need to sit down and discuss this feature together :)